### PR TITLE
[Python] TLV schema tag should start from 1

### PR DESCRIPTION
**Pull request by erjiaqing**
#### Problem

TLV schema tag should start from 1, and the error is not captured.

#### Change overview

Update codegen, and add tests to verify if we successfully decoded all values.

Create `ValueDecodeFailure` class to wrap the value, since exceptions will be ignored in C callbacks.

Wrap tests by _AssumeDecodeSuccess, users don't have to do this manually, they will see ValueDecodeFailure instead of the attribute describer.

Adds a new global debug mode variable that can be set using `mattersetdebug()` to facilitate debugging of issues (e.g throwing exceptions in the relevant modules instead of returning formatted ValueDecodeFailure objects).

#### Testing

CI with first commit only should fail: https://github.com/erjiaqing/connectedhomeip/actions/runs/1519033903
CI with both commits should success.

### CODE REVIEW

**Review by woody-apple — APPROVED** [review]
Fast tracking given this is only updating testing tools for developers. Would be great to update this for CSG.

File: src/controller/python/chip/clusters/Attribute.py [hunk 1]
```diff
@@ -199,21 +205,24 @@ def _handleAttributeData(self, path: AttributePathWithListIndex, status: int, da
...
-                    attributeValue = attributeType.FromTLV(data)
-                except:
+                    attributeValue = attributeType(attributeType.FromTLV(data))
+                except Exception as ex:
                     logging.error(
                         f"Error convering TLV to Cluster Object for path: Endpoint = {path.EndpointId}/Cluster = {path.ClusterId}/Attribute = {path.AttributeId}")
                     logging.error(
                         f"Failed Cluster Object: {str(attributeType)}")
-                    raise
+                    attributeValue = ValueDecodeFailure(
```
Comment (mrjerryjohns) [inline src/controller/python/chip/clusters/Attribute.py:220]: I'd very much rather pass the exception through - wrapping it like this circumvents the exception machinery.

**Review by mrjerryjohns — CHANGES_REQUESTED** [review]

File: src/controller/python/chip/clusters/Attribute.py (hunk 1 above)
Comment (mrjerryjohns) [inline src/controller/python/chip/clusters/Attribute.py:220]: Also, I don't think it's necessary to pass the raw TLV through - that's just noisy.

We already print out a lot of valuable information about what went wrong in the lines above, that should be sufficient.

**Review by mrjerryjohns — APPROVED** [review]
As per discussions with @erjiaqing, I have come around to the approach of not throwing exceptions up to the user, but rather, returning the result as ValueDecodeFailure objects that in turn, contain the exception. This permits retrieving all attribute data without terminally failing the whole interaction on a single bad attribute.

However, this does make debugging issues more difficult. I've gone ahead and added a `mattersetdebug()` function that has a global built-in `enableDebugMode` variable that can be used to raise exceptions instead when trying to debug these failures. That will in turn, return the exception traceback which is needed to debug stuff.

**Review by yufengwangca — APPROVED** [review]

**Pull request merged**